CMakeLists.txt

```diff
@@ -12,27 +12,21 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-cmake_minimum_required(VERSION 2.8.12)
-if (POLICY CMP0048)
-  cmake_policy(SET CMP0048 NEW)
-endif()
-if (POLICY CMP0054)
-  # Avoid dereferencing variables or interpret keywords that have been
-  # quoted or bracketed.
-  # https://cmake.org/cmake/help/v3.1/policy/CMP0054.html
-  cmake_policy(SET CMP0054 NEW)
-endif()
+cmake_minimum_required(VERSION 3.0)
+project(spirv-tools VERSION 2019.5 LANGUAGES CXX)
```

Comment:
Saying`Version 2019.15` sets PROJECT_VERSION* variables. This is used below to write the SPIRV-ToolsConfigVersion.cmake file.   (at https://github.com/KhronosGroup/SPIRV-Tools/pull/2784/files#diff-af3b638bc2a3e6c650974192a53c7291R310)

Currently, all versioning for the project is controlled by the first lines in the CHANGES file.  (See https://github.com/KhronosGroup/SPIRV-Tools#versioning-spirv-tools)   The implementation is in utils/update_build_version.py function deduce_software_version.

So this change introduces a new source of truth for versioning.   How does it interact with our conventions for the -dev versions being non-final? 

Can we instead set that variable by running a command that parses the CHANGES file?
E.g.  by using an execute_command, as suggested here: https://stackoverflow.com/questions/13037230/how-do-i-initialize-a-cmake-variable-with-the-result-of-a-shell-command

Reply:
Oh I wasn't aware of how `CHANGES` is used. Parsing that file should be possible.